frontend/src/metabase/css/dashboard.css

```
@@ -101,6 +101,14 @@
   z-index: 2;
 }
 
+/* dashcards use positioned absolutely,
+   so children calculate z-index inside a container, not globally,
+   that's why we have to override parent's z-index
+*/
```

I feel like this comment makes sense only given the full context of the PR and knowing previous attempts to solve the problem. The fix is straightforward, I don't need this comment to understand it.

> so children calculate z-index inside a container, not globally,

A nicer way to put this: "so it forms a new [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context)".



```suggestion
/**
 * Dashcards are positioned absolutely so each one forms a new stacking context.
 * The dashcard user is currently interacting with needs to be positioned above other dashcards
 * to make sure it's not covered by absolutely positioned children of neighboring dashcards.
 *
 * @see https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context
 */
```

---

great improvement, thanks!